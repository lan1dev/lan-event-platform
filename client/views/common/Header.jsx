import styled from 'styled-components';
import colors from '../constants/colors';

/* eslint-disable react/destructuring-assignment */
const Header = styled.header`
  background: ${props => (props.secondary ? colors.secondary : colors.white)};
  padding: 0 50px;
  height: 64px;
  line-height: 64px;
  flex: 0 0 auto;
  box-sizing: border-box;
`;
/* eslint-enable react/destructuring-assignment */

export default Header;
