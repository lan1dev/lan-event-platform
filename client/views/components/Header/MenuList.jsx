import styled from 'styled-components';

const MenuList = styled.ul`
  list-style-type: none;
  margin: 0 0 0 50px;
  padding: 0;

  & li {
    float: left;
    margin: 0 10px;
  }
`;

export default MenuList;
