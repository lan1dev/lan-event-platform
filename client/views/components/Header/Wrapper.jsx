import styled from 'styled-components';

const Wrapper = styled.div`
  height: 50px;
  width: 100%;
  border-bottom: 1px solid black;
  line-height: 50px;
  display: flex;
  justify-content: space-between;
  background: white;
  padding: 0 20px;
`;

export default Wrapper;
