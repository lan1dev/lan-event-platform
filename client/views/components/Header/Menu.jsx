import React, { Fragment } from 'react';
import PropTypes from 'prop-types';
import { Link } from 'react-router-dom';
import MenuList from './MenuList';

const propTypes = {
  items: PropTypes.arrayOf(PropTypes.shape({ name: PropTypes.string, href: PropTypes.string }))
    .isRequired
};

const Menu = ({ items }) => (
  <Fragment>
    <MenuList>
      {items.map(item => (
        <li>
          <Link key={item.href} to={item.href}>
            {item.name}
          </Link>
        </li>
      ))}
    </MenuList>
  </Fragment>
);

Menu.propTypes = propTypes;

export default Menu;
