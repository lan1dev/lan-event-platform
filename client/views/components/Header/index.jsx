import React from 'react';
import { Link } from 'react-router-dom';
import Wrapper from './Wrapper';
import menuConfig from '../../../configs/menuConfig';
import Menu from './Menu';
import FlexBox from '../../common/FlexBox';

const Header = () => (
  <Wrapper>
    <FlexBox>
      <Link to="/">Logo</Link>
      <Menu items={menuConfig} />
    </FlexBox>
    <div>
      <Link to="/login">Login</Link> / <Link to="Register">Register</Link>
    </div>
  </Wrapper>
);

export default Header;
