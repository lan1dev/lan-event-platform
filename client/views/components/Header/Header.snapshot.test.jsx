import React from 'react';
import { shallow } from 'enzyme';
import toJson from 'enzyme-to-json';
import 'jest-styled-components';
import Header from './index';

describe('Header', () => {
  let props;
  let wrapper;

  const Wrapper = () => {
    if (!wrapper) {
      wrapper = shallow(<Header {...props}>{props.children}</Header>);
    }
    return wrapper;
  };

  // Reset Component and Props before each test.
  beforeEach(() => {
    props = {};
    wrapper = undefined;
  });

  describe('without props', () => {
    it('Should match snapshot', () => {
      expect(toJson(Wrapper())).toMatchSnapshot();
    });
  });

  describe('with props', () => {
    it('Should match snapshot', () => {
      props = {
        primary: true
      };
      expect(toJson(Wrapper())).toMatchSnapshot();
    });
  });
});
