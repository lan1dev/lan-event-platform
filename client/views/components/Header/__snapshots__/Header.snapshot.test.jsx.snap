// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Header with props Should match snapshot 1`] = `
.c0 {
  height: 50px;
  width: 100%;
  border-bottom: none;
  line-height: 50px;
  display: -webkit-box;
  display: -webkit-flex;
  display: -ms-flexbox;
  display: flex;
  -webkit-box-pack: justify;
  -webkit-justify-content: space-between;
  -ms-flex-pack: justify;
  justify-content: space-between;
  background: white;
  padding: 0 20px;
}

<header
  className="c0"
/>
`;

exports[`Header without props Should match snapshot 1`] = `
.c0 {
  height: 50px;
  width: 100%;
  border-bottom: 1px solid black;
  line-height: 50px;
  display: -webkit-box;
  display: -webkit-flex;
  display: -ms-flexbox;
  display: flex;
  -webkit-box-pack: justify;
  -webkit-justify-content: space-between;
  -ms-flex-pack: justify;
  justify-content: space-between;
  background: white;
  padding: 0 20px;
}

<header
  className="c0"
/>
`;
